astropy/table/column.py

```diff
@@ -706,6 +706,11 @@ def __new__(cls, data=None, name=None,
                                           copy=copy, copy_indices=copy_indices)
         return self
 
+    def __setattr__(self, item, value):
+        if not isinstance(self, MaskedColumn) and item == "mask":
+            raise AttributeError("Cannot set mask value to a Column in non-masked Table")
```

Comment:
Use lower case to start the error message, i.e. `cannot set ..`.  We are not consistent about this but new code should follow that convention.